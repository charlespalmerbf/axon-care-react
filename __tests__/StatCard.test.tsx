import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import StatCard from "../components/StatCard";

describe("StatCard", () => {
  it("should render with appointments type", () => {
    render(
      <StatCard
        type="appointments"
        count={10}
        label="Appointments"
        icon="/icons/appointments.svg"
      />,
    );

    expect(screen.getByText("10")).toBeInTheDocument();
    expect(screen.getByText("Appointments")).toBeInTheDocument();
    expect(screen.getByAltText("Appointments")).toBeInTheDocument();
    expect(screen.getByText("Appointments").closest("div")).toHaveClass(
      "bg-appointments",
    );
  });

  it("should render with pending type", () => {
    render(
      <StatCard
        type="pending"
        count={5}
        label="Pending"
        icon="/icons/pending.svg"
      />,
    );

    expect(screen.getByText("5")).toBeInTheDocument();
    expect(screen.getByText("Pending")).toBeInTheDocument();
    expect(screen.getByAltText("Pending")).toBeInTheDocument();
    expect(screen.getByText("Pending").closest("div")).toHaveClass(
      "bg-pending",
    );
  });

  it("should render with cancelled type", () => {
    render(
      <StatCard
        type="cancelled"
        count={3}
        label="Cancelled"
        icon="/icons/cancelled.svg"
      />,
    );

    expect(screen.getByText("3")).toBeInTheDocument();
    expect(screen.getByText("Cancelled")).toBeInTheDocument();
    expect(screen.getByAltText("Cancelled")).toBeInTheDocument();
    expect(screen.getByText("Cancelled").closest("div")).toHaveClass(
      "bg-cancelled",
    );
  });
});
