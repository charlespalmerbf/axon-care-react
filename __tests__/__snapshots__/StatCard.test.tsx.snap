// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatCard matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="stat-card bg-appointments"
      >
        <div
          class="flex items-center gap-4"
        >
          <img
            alt="Appointments"
            class="size-8 w-fit"
            data-nimg="1"
            decoding="async"
            height="32"
            loading="lazy"
            src="/_next/image?url=%2Ficon.png&w=64&q=75"
            srcset="/_next/image?url=%2Ficon.png&w=32&q=75 1x, /_next/image?url=%2Ficon.png&w=64&q=75 2x"
            style="color: transparent;"
            width="32"
          />
          <h2
            class="text-32-bold text-white"
          >
            10
          </h2>
        </div>
        <p
          class="text-14-regular"
        >
          Appointments
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="stat-card bg-appointments"
    >
      <div
        class="flex items-center gap-4"
      >
        <img
          alt="Appointments"
          class="size-8 w-fit"
          data-nimg="1"
          decoding="async"
          height="32"
          loading="lazy"
          src="/_next/image?url=%2Ficon.png&w=64&q=75"
          srcset="/_next/image?url=%2Ficon.png&w=32&q=75 1x, /_next/image?url=%2Ficon.png&w=64&q=75 2x"
          style="color: transparent;"
          width="32"
        />
        <h2
          class="text-32-bold text-white"
        >
          10
        </h2>
      </div>
      <p
        class="text-14-regular"
      >
        Appointments
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
